lib/src/state/strand_maker.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:scadnano/src/state/design.dart';
+import 'package:scadnano/src/state/domain.dart';
+import 'package:scadnano/src/state/loopout.dart';
+import 'package:scadnano/src/state/modification.dart';
+import 'package:scadnano/src/state/strand.dart';
+import 'package:color/color.dart';
+import 'package:scadnano/src/state/substrand.dart';
+
+class StrandMaker {
```

Comment:
Did you see this comment?

https://github.com/UC-Davis-molecular-computing/scadnano/pull/411#pullrequestreview-459602804

> Internally, the StrandMaker object should keep a reference to the Design that created it (in fact, I'm picturing that the only way to create a StrandMaker object is by calling Design.strand, although I don't think we need to enforce that), so that when commit() is called, it knows to modify that Design to create the new Design with the new Strand.

In particular, the `commit()` method should return the new `Design` (instead of returning the `Strand`).